mod-source-record-manager-server/src/main/java/org/folio/services/afterprocessing/HrIdFieldServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package org.folio.services.afterprocessing;
+
+import org.folio.rest.jaxrs.model.Record;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Objects;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.folio.services.afterprocessing.AdditionalFieldsUtil.addDataFieldToMarcRecord;
+import static org.folio.services.afterprocessing.AdditionalFieldsUtil.getValue;
+import static org.folio.services.afterprocessing.AdditionalFieldsUtil.isFieldExist;
+
+@Service
+public class HrIdFieldServiceImpl implements HrIdFieldService {
+
+  private static final String TAG_001 = "001";
+  private static final String TAG_003 = "003";
+  private static final String TAG_035 = "035";
+  private static final char SUBFIELD_FOR_035 = 'a';
+  private static final char INDICATOR_FOR_035 = ' ';
+
+  @Override
+  public void move001valueTo035Field(List<Record> records) {
+    records.stream().parallel().forEach(record -> {
+      String valueFrom035 = getValue(record, TAG_035, SUBFIELD_FOR_035);
+      if (Objects.isNull(valueFrom035)) {
```

Comment:
So this line is the culprit for matching not working for records with multiple 035. Probably need to add the same logic as in SRS for adding 035 but not duplicating 035s.